FeedStoreChallenge/CoreDataFeedStore.swift

```diff
@@ -29,14 +29,80 @@ public final class CoreDataFeedStore: FeedStore {
 	}
 
 	public func retrieve(completion: @escaping RetrievalCompletion) {
-		fatalError("Must be implemented")
+		context.perform { [context] in
+			do {
+				if let existingCache = try ManagedCache.fetch(in: context) {
+					completion(.found(feed: existingCache.feed.toLocal(), timestamp: existingCache.timestamp))
+				} else {
+					completion(.empty)
+				}
+			} catch {
+				completion(.failure(error))
+			}
+		}
 	}
 
 	public func insert(_ feed: [LocalFeedImage], timestamp: Date, completion: @escaping InsertionCompletion) {
-		fatalError("Must be implemented")
+		context.perform { [context] in
+			do {
+				let existingCache = try ManagedCache.fetch(in: context)
+				existingCache.map(context.delete)
```

Comment:
This logic is duplicated between methods `insert` and `delete` (lines 67 and 68). Also, since `fetch` returns an optional `ManagedCache`, the name `existingCache` is misleading since it can be `nil`. 
We can extract this logic to a helper method, or inline it as a single operation making use of optional `map`:
```suggestion
				try ManagedCache.fetch(in: context).map(context.delete)
```